### PR TITLE
chore(providers): rename 42School to FortyTwo

### PR DESCRIPTION
**Pull request by ikersojo**
## ☕️ Reasoning

**42School Provider** works but documentation was reflecting a different providerName. In fact, the providerName had spaces and starts with numbers. I have replaced all instances of the providerName so that documentation and the getting started guides reflect the correct providerName.

Changed from `42School` or `42 School` in the documentation to `FortyTwo` in this PR.

## 🧢 Checklist

- [x] Documentation
- [ ] Tests
- [ ] Ready to be merged

## 📌 Resources

- [Security guidelines](https://github.com/nextauthjs/.github/blob/main/SECURITY.md)
- [Contributing guidelines](https://github.com/nextauthjs/.github/blob/main/CONTRIBUTING.md)
- [Code of conduct](https://github.com/nextauthjs/.github/blob/main/CODE_OF_CONDUCT.md)
- [Contributing to Open Source](https://kcd.im/pull-request)

### CODE REVIEW

**Comment by k-taro56** [comment]
Duplicate of #10980

**Comment by ndom91** [comment]
@k-taro56 thanks for that big PR! I left a littel note there.

After we cleanup the `42School` imports, etc. there though, we can continue in this PR with any  required docs changes to make everything match again (i.e. in `docs/pages/getting-started/providers/42-school.mdx` potentially). Cool with everyone?

@ikersojo I'll ping you here once that other PR is merged and then we'd appreciate it if you could modify this to make the docs match any code changes :pray:

**Comment by ikersojo** [comment]
Sure, happy to help!

**Comment by ndom91** [comment]
> Sure, happy to help!

We've merged that PR, would love if you could clean this up and we can ship the rest of your 42school fixes 👍

**Comment by ikersojo** [comment]
42 doc should be now aligned

File: packages/core/src/providers/42-school.ts [hunk 1]
```diff
@@ -186,17 +186,17 @@ export interface FortyTwoProfile extends UserData, Record<string, any> {
  *
  * #### Configuration
  *```ts
- * import { Auth } from "@auth/core"
- * import FortyTwoSchool from "@auth/core/providers/42-school"
+ * import Auth from "@auth/core"
```
Comment (ndom91) [inline packages/core/src/providers/42-school.ts:189]: Small note here, there is no default export, so the preexisting named export here was correct

**Review by ndom91 — CHANGES_REQUESTED** [review]

File: packages/core/src/providers/42-school.ts (hunk 1 above)
Comment (ikersojo) [reply packages/core/src/providers/42-school.ts:189]: Sorry, my bad on the back and forward with the commits. Well spotted!
Thanks

File: packages/core/src/providers/42-school.ts [hunk 2]
```diff
@@ -186,17 +186,17 @@ export interface FortyTwoProfile extends UserData, Record<string, any> {
...
- *     FortyTwoSchool({
- *       clientId: FORTY_TWO_SCHOOL_CLIENT_ID,
- *       clientSecret: FORTY_TWO_SCHOOL_CLIENT_SECRET,
- *     }),
- *   ],
+       FortyTwo({
+         clientId: 42_SCHOOL_CLIENT_ID,
+         clientSecret: 42_SCHOOL_CLIENT_SECRET
+       })
+     ],
```
Comment (k-taro56) [inline packages/core/src/providers/42-school.ts:199]:
```suggestion
 *     FortyTwo({
 *       clientId: FORTY_TWO_CLIENT_ID,
 *       clientSecret: FORTY_TWO_CLIENT_SECRET,
 *     }),
 *   ],
```

Comment (ikersojo) [reply packages/core/src/providers/42-school.ts:199]: I would not recommend to do so unless you are willing to replace the `42-School` by `FortyTwo` everywhere else in the code base, even the file names 
Rationale: Using `Forty_Two` (or `FortyTwo` for consistency with current provider name) can be a good idea to remove numbers from the naming scheme. However, the 42-school reference is used across the codebase.
An example
```ts
//docs/pages/data/manifest.json
  ...
  "providersOAuth": {
    "github": "GitHub",
    "google": "Google",
    "twitter": "Twitter",
    "apple": "Apple",
    "discord": "Discord",
    "auth0": "Auth0",
    "facebook": "Facebook",
    "keycloak": "Keycloak",
    "42-school": "FortyTwo", // <-- // used in docs/components/OAuthProviderInstructions/content/index.tsx, for example
```

I am happy to do it myself. @ndom91 which way should I proceed?

Comment (k-taro56) [reply packages/core/src/providers/42-school.ts:199]: I agree.

But this is TypeScript code, so we can't use variables that start with a number.

And at least asterisks are required.

Comment (ndom91) [reply packages/core/src/providers/42-school.ts:199]: Yeah so I think refactoring this to use `FortyTwo` everywhere makes sense. But as yuo rightly pointed out, it is used all over the place. 

We'd really appreciate if yuo cuold go through this with a fine tooth comb and make sure its not breaking anything :blush:

**Comment by ikersojo** [comment]
I think everything is now consistent.

**Review by balazsorban44 — COMMENTED** [review]
I am not sure about this one. Besides being a **breaking** change, the official name seems to be 42 school, using the number format: https://42.fr/en/homepage/

Is there a reason for this change to be necessary?

**Comment by ikersojo** [comment]
It was suggested by @ndom91 to align everything to 'FortyTwo'. Since it was already a breaking change, I am proposing to go for 'FortyTwoSchool', as a mix between both naming schemes.

We can revert the changes to the naming scheme. As said before, this PR was a documentation fix, but as a 42 student, I am happy to help. Please let me know how to proceed.

**Comment by balazsorban44** [comment]
My main concern is with the import path change: `42-school` -> `fortyTwoSchool` ❌. This seems unnecessary, breaking, and not aligned with the case convention.

42School -> FortyTwoSchool seems also wrong, searching for 42 in our docs would not work anymore.

I'll close this for now as I think this would be confusing to current users, breaking for a name change.

**Pull request closed without merging**